internal/commands/server/k8s_e2e_test.go

```diff
@@ -1036,6 +1036,45 @@ func TestHTTPMeshService(t *testing.T) {
 	testenv.Test(t, feature.Feature())
 }
 
+func TestGatewayWithConsulNamespaceDoesntExist(t *testing.T) {
+	feature := features.New("gateway class in k8s namespace where consul namespace not created yet, consul enterprise").
+		Assess("gateway is created with appropriate number of replicas set", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+
+			namespace := e2e.Namespace(ctx)
+			client := e2e.ConsulClient(ctx)
+
+			if e2e.IsEnterprise() {
```

Comment:
Should we only run this test when we're running enterprise? for the oss version, this test isn't really doing anything, right?

I guess it's not really doing any harm, but I could see it being confusing needing to work in this test code one day if it's not clear that the test is only exercised for enterprise.

Reply:
I ran into a case where when the code path attempted to run on the non enterprise version, I was getting nil pointer exceptions so I initially let it run for both, but I'm also ok just skipping the test for non enterprise since that path should be covered by the other tests.